packages/stryker/src/utils/TempFolder.ts

```diff
@@ -0,0 +1,89 @@
+import * as fs from 'mz/fs';
+import * as path from 'path';
+import * as mkdirp from 'mkdirp';
+import * as log4js from 'log4js';
+import { deleteDir } from './fileUtils';
+
+const log = log4js.getLogger('TempFolder');
+
+export class TempFolder {
+  baseTempFolder: string;
+  tempFolder: string;
+  
+  private constructor() { }
+
+  initialize(tempDirName = '.stryker-tmp') {
+    this.baseTempFolder = path.join(process.cwd(), tempDirName);
+    this.tempFolder = path.join(this.baseTempFolder, this.random().toString());
+    mkdirp.sync(this.baseTempFolder);
+    mkdirp.sync(this.tempFolder);
+  }
+
+  /**
+   * Creates a new random folder with the specified prefix.
+   * @param prefix The prefix.
+   * @returns The path to the folder.
+   */
+  createRandomFolder(prefix: string): string {
+    let dir = this.baseTempFolder + path.sep + prefix + this.random();
+    mkdirp.sync(dir);
+    return dir;
+  }
+
+  /**
+   * Writes data to a specified file.
+   * @param filename The path to the file.
+   * @param data The content of the file.
+   * @returns A promise to eventually save the file.
+   */
+  writeFile(filename: string, data: string): Promise<void> {
+    return fs.writeFile(filename, data, { encoding: 'utf8' });
+  }
+
+  /**
+   * Copies a file.
+   * @param fromFilename The path to the existing file.
+   * @param toFilename The path to copy the file to.
+   * @param instrumenter An optional additional instrumenter to stream the file through
+   * @returns A promise to eventually copy the file.
+   */
+  copyFile(fromFilename: string, toFilename: string, instrumenter: NodeJS.ReadWriteStream | null): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+      let readStream: NodeJS.ReadableStream = fs.createReadStream(fromFilename, { encoding: 'utf8' });
+      let writeStream = fs.createWriteStream(toFilename);
+      readStream.on('error', reject);
+      writeStream.on('error', reject);
+      if (instrumenter) {
+        readStream = readStream.pipe(instrumenter);
+      }
+      readStream.pipe(writeStream);
+      readStream.on('end', () => resolve());
+    });
+  }
+
+  /**
+   * Deletes the Stryker-temp folder
+   */
+  clean() {
+    log.debug(`Cleaning stryker temp folder ${this.baseTempFolder}`);
```

Comment:
Is word "clean" in that context"? As I understand "clean" means "clean contents of the directory". But purpose of that function is to remove directory completely.

Are you sure this is correct? Perhaps "remove" will be better since it will say what really happens.

Reply:
Yeah, `remove` would be better. Yes the purpose is to remove the dir completely.

Reply:
> Is word "clean" in that context"?

Sorry I wanted to say "is it correct in that context"?